core/templates/dev/head/components/ExplorationCreationService.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Modal and functionality for the create exploration button.
+ */
+
+// Service for the create/upload exploration buttons and modals.
+oppia.factory('ExplorationCreationService', [
+  '$http', '$modal', '$timeout', '$rootScope', '$window',
+  'alertsService', 'siteAnalyticsService', 'urlService',
+  function(
+      $http, $modal, $timeout, $rootScope, $window,
+      alertsService, siteAnalyticsService, urlService) {
+    return {
+      createNewExploration: function() {
+        alertsService.clearWarnings();
+
+        var currentPathname = urlService.getPathname();
+
+        if (currentPathname !== '/my_explorations') {
+          $window.location.replace('/my_explorations?mode=create');
+        } else {
+          siteAnalyticsService.registerOpenExplorationCreationModalEvent();
```

Comment:
I feel like this warrants new method names and/or documentation changes. That's not really what's being recorded anymore.